main.go

```diff
@@ -78,21 +80,66 @@ func main() {
 	} else if ctx.SetProtocol(gpgme.ProtocolAssuan) != nil {
 		log.Debugf("Cannot initialize Assuan IPC: %v. Disabling GPG and SSH watchers.", err)
 	} else {
-		gpgPubringPath := path.Join(gpgme.GetDirInfo("homedir"), "pubring.kbx")
-		if _, err := os.Stat(gpgPubringPath); err == nil {
-
-			requestGPGCheck := make(chan bool)
-			go detector.CheckGPGOnRequest(requestGPGCheck, notifiers, ctx)
-			go detector.WatchGPG(gpgPubringPath, requestGPGCheck)
-			go detector.WatchSSH(requestGPGCheck, exits)
-		} else {
-			log.Debugf("'%v' could not be found. Disabling GPG and SSH watchers.", gpgPubringPath)
+		var gpgPubringPath = path.Join(gpgme.GetDirInfo("homedir"), "private-keys-v1.d")
+		if _, err := os.Stat(gpgPubringPath); os.IsNotExist(err) {
+			fmt.Printf("Directory '%s' does not exist (you have no private keys).\n", gpgPubringPath)
+			return
 		}
+		var searchTerm = "shadowed-private-key"
+		var filesToWatch []string
+		filesToWatch, err := findMatchingFiles(gpgPubringPath, searchTerm)
+		if err != nil {
+			fmt.Printf("Error finding files: %v\n", err)
+			return
+		}
+		if len(filesToWatch) == 0 {
+			fmt.Printf("No files matching the term '%s' found.\n", searchTerm)
+			return
+		}
+		requestGPGCheck := make(chan bool)
+		go detector.CheckGPGOnRequest(requestGPGCheck, notifiers, ctx)
+		go detector.WatchGPG(filesToWatch, requestGPGCheck)
+		go detector.WatchSSH(requestGPGCheck, exits)
 	}
 	wait := make(chan bool)
 	<-wait
 }
 
+func findMatchingFiles(folderPath, term string) ([]string, error) {
+	var result []string
+	err := filepath.Walk(folderPath, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if info.IsDir() {
+			return nil
+		}
+		file, err := os.Open(path)
+		if err != nil {
+			return err
+		}
+		defer file.Close()
+
+		scanner := bufio.NewScanner(file)
```

Comment:
I'd also go for something simpler, the files are small anyway, no need to optimize reading them so much, how does something like this look to you?


```go
		if err != nil || info.IsDir() {
			return err
		}
		data, err := os.ReadFile(path)
		if err != nil {
			return err
		}
		if strings.Contains(string(data), term) {
			result = append(result, path)
		}
		return nil
```